### ISSUE TITLE
Add hook support

### ISSUE THREAD

**Issue by diegobernardes** [rubenv/sql-migrate#120]

**Comment by rubenv** [comment]
Hi, could you explain what you mean by this?

**Comment by diegobernardes** [comment]
Yes, sorry, clicked in the create by mistake, 1 sec, writing the message.

**Comment by rubenv** [comment]
I like the idea of hooks and I don't mind that they might not be available through the CLI.

If we do add them it's probably best to scan for them when using the CLI and refuse to operate when a hook is detected. Just to make sure things either happen correctly or not at all.

There's a real danger for abuse though, so they should be used sparingly.
